File: Objective_C/autosObectiveC/autosObectiveC/auto.m
//
//  auto.m
//  autosObectiveC
//
//  Created by Elías Jiménez on 23/08/24.
//

#import "auto.h"

@implementation Auto

- (id) initSinP: (NSString*) marca modelo: (NSString*) modelo {
    self = [super init];
    _marca = marca;
    _modelo = modelo;
    _posicion = 0;
    return self;
}

- (void) mover {
    _posicion += 1;
}

- (void) moverX: (int) x {
    _posicion += x;
}

- (void) mostrar {
    NSLog(@"%@ - %@ | %d Km", _marca, _modelo, _posicion);
}

@end

File: swift/AutosStoryBoards/AutosStoryBoards/Auto.swift
//
//  Auto.swift
//  AutosStoryBoards
//
//  Created by Elías Jiménez on 02/09/24.
//

import Foundation

class Auto {
    var marca: String
    var modelo: String
    var posicion: Int
    
    init() {
        marca = ""
        modelo = ""
        posicion = 0
    }
    
    init(marca:String, modelo:String) {
        self.modelo = modelo
        self.marca = marca
        self.posicion = 0
    }
    
    init(marca:String, modelo:String, posicion:Int) {
        self.modelo = modelo
        self.marca = marca
        self.posicion = posicion
    }
    
    func avanzar() {
        posicion += 1
    }
    
    func avanzar(x: Int) {
        posicion += x
    }
    
    func imprimir() -> String {
        return "\(marca) \(modelo) | \(posicion) Km"
    }
}
